tests/plugins/test_solvers.py

```diff
@@ -62,6 +70,17 @@ def test_get_cached_solver_backend(plugin_manager, mocker):
     assert mocked.call_count == 1  # real caching!
 
 
+def test_solver_user_agent(plugin_manager, monkeypatch):
+    plugin = VerboseSolverPlugin()
+    plugin_manager.register(plugin)
+    monkeypatch.setattr(context, "solver", "verbose-classic")
+    # context.user_agent is a memoizedproperty, which may have been cached from
+    # previous test runs. We're checking here and clear the cache if needed.
+    if "__user_agent" in context._cache_:
+        del context._cache_["__user_agent"]
+    assert verbose_user_agent in context.user_agent
```

Comment:
Should we clear the cache afterwards too?

Reply:
yep, https://github.com/conda/conda/pull/12104/commits/c74065797f87e80c567bb0ab5962ef6ad8eb5838